### ISSUE TITLE
FlatLAF is breaking JOSM

### ISSUE THREAD

**Issue by mikkolukas** [JFormDesigner/FlatLaf#611]
The FlatLAF plugin for JOSM directs me to https://www.formdev.com/flatlaf/ for further info, which directs me to here.
**Please tell me where to go, if this is not the right place.**

**What I do:**

1. Install FlatLAF in JOSM
2. Restart JOSM (which works fine)
3. Select the FlatLaf Darcula theme
4. Restart JOSM (or so I thought)

**Expected behavior:**

JOSM coming back from restart

**Actual behavior:**

JOSM does not restart and I am not able to start it manually again. It just dies within one second from launching, not even showing the loading box.

&#x200B;

**Workaround:**

1. Start JOSM with the `--skip-plugins` parameter like this (in Windows):%JAVA\_HOME%\\bin\\javaw.exe -jar josm-tested.jar --skip-plugins
2. JOSM starts up
   1. (just an unimportant side note: Weirdly enough, with some alternative theme loaded. If I close JOSM and start it a second time with the `--skip-plugins` parameter, it will load the default JOSM theme. No big worries, we don't have to restart it such a second time for the solution to work)
3. Close JOSM.
4. Start JOSM normally (without the --skip-plugins parameter)
5. Uninstall FlatLAF

**Context:**

JOSM Version 18583

Same problem occurs on a completely newly installed JOSM without any cache, preferences, plugins.

I am using the JOSM parameters `-Djosm.home=.` `-Djosm.pref=./preferences` `-Djosm.cache=./cache` as documented [here](https://josm.openstreetmap.de/wiki/Help/CommandLineOptions#Javaoptions), as I want to have those close by and my JOSM as a stand alone application

**Comment by DevCharly** [comment]
Thanks for reporting.

This is not a bug in FlatLaf.

It is an issue in the JOSM FlatLaf plugin (only dark themes) **version 36034**. Older version work.
Someone updated the used FlatLaf version from 1.6.1 to 2.6 without fixing an incompatibility.

When running JOSM from command line it outputs following error:
~~~
INFO: loading plugin 'flatlaf' (version 36034)
variable or property '@textComponentBackground' not found
~~~

`@textComponentBackground` has been renamed to `@componentBackground` in FlatLaf 2.0.

I'll fix the JOSM FlatLaf plugin...

**Comment by DevCharly** [comment]
The issue is fixed in JOSM flatlaf plugin **version 36036**.

**Comment by tsmock** [comment]
Sorry about that -- my fault.

I didn't see any exceptions when I built the new `flatlaf` version, so I assumed (wrongly) that everything was OK.

Is there a way to validate the properties file via the command line, or is it something that needs to be loaded?

**Comment by mikkolukas** [comment]
@DevCharly **Thank you for the fast response and fix! Highly appreciated!** 🙂

Is it possible that the reference to documentation for the plugin can be changed so it doesn't lead to here?
I tried my best to find a place to report for bugs in the plugin (as I rightly assumed was separate from FlatLAF itself), but couldn't find it.

**Comment by tsmock** [comment]
@mikkolukas : I don't know which documentation you are referring to, but as of [r18598](https://josm.openstreetmap.de/changeset/18598/josm/), if there is a problem with the LaF, JOSM will set the Look and Feel to the platform default and open a standard bug report window which will allow users to open a ticket at JOSM trac.

**Comment by mikkolukas** [comment]
@tsmock Inside JOSM, where you install plugins, you'll find the JOSM FlatLaf plugin listed as shown here:

![image](https://user-images.githubusercontent.com/1522239/202889923-d2964544-e697-4e17-905b-3b18c128ffd3.png)

The link leads directly to [https://www.formdev.com/flatlaf/](https://www.formdev.com/flatlaf/), misleading one to:

1. Believe that FlatLaf and the JOSM FlatLaf plugin is the one and same thing (I suspected it was not).
2. Not have a proper place to report errors that are related to the JOSM FlatLaf plugin, but not FlatLaf itself (I did a good effort to google for it, but couldn't find it anywhere).

Hence the situation we are now in, that I reported the bug to FlatLaf, while the error was in the JOSM FlatLaf plugin, bothering the upstream developer.
